distribution/test/itests/test-itests-ddf/src/test/java/ddf/test/itests/catalog/TestFederation.java

```diff
@@ -1301,6 +1302,12 @@ public void testFederatedDownloadProductToCacheOnlyCacheEnabled() throws Excepti
                 .body(is(String.format("The product associated with metacard [%s] from source [%s] is being downloaded to the product cache.", metacardId, CSW_SOURCE_ID)));
         // @formatter:on
 
+        expect("Cached file exists").within(10, SECONDS).until(() -> Files.exists(Paths.get(ddfHome)
+                .resolve(PRODUCT_CACHE).resolve(CSW_SOURCE_ID + "-" + metacardId)));
+        expect("Cached metacard exists").within(10, SECONDS)
```

Comment:
Instead of checking for the file existence, maybe we should wait until the CometD events work properly and use those?